x-pack/plugins/security_solution/cypress/e2e/detection_rules/rule_actions.cy.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getIndexConnector } from '../../objects/connector';
+import { getSimpleRule } from '../../objects/rule';
+import { NUMBER_OF_ALERTS, ALERT_GRID_CELL } from '../../screens/alerts';
+
+import { goToRuleDetails } from '../../tasks/alerts_detection_rules';
+import { createEmptyDocument, createIndex, deleteIndex } from '../../tasks/api_calls';
+import {
+  cleanKibana,
+  deleteAlertsAndRules,
+  deleteConnectors,
+  deleteDataView,
+  postDataView,
+} from '../../tasks/common';
+import {
+  createAndEnableRule,
+  fillAboutRuleAndContinue,
+  fillDefineCustomRuleAndContinue,
+  fillRuleAction,
+  fillScheduleRuleAndContinue,
+  waitForAlertsToPopulate,
+  waitForTheRuleToBeExecuted,
+} from '../../tasks/create_new_rule';
+import { login, visit } from '../../tasks/login';
+
+import { RULE_CREATION } from '../../urls/navigation';
+
+describe('Rule actions', () => {
+  const INDEX_CONNECTOR = getIndexConnector();
+
+  before(() => {
+    cleanKibana();
+    login();
+    createIndex(INDEX_CONNECTOR.index);
+    createEmptyDocument(INDEX_CONNECTOR.index);
+    postDataView(INDEX_CONNECTOR.index);
+  });
+
+  beforeEach(() => {
+    deleteAlertsAndRules();
+    deleteConnectors();
+  });
+
+  after(() => {
+    deleteIndex(INDEX_CONNECTOR.index);
+    deleteDataView(INDEX_CONNECTOR.index);
+  });
+
+  const rule = {
+    ...getSimpleRule(),
+    actions: { interval: 'rule', connectors: [INDEX_CONNECTOR] },
+  };
+
+  it('Creates a custom query rule with an index action ', function () {
+    visit(RULE_CREATION);
+    fillDefineCustomRuleAndContinue(rule);
+    fillAboutRuleAndContinue(rule);
+    fillScheduleRuleAndContinue(rule);
+    fillRuleAction(rule);
+    createAndEnableRule();
+    goToRuleDetails();
+    waitForTheRuleToBeExecuted();
+    waitForAlertsToPopulate();
+
+    cy.get(NUMBER_OF_ALERTS)
+      .invoke('text')
+      .should('match', /^[1-9].+$/);
+    cy.get(ALERT_GRID_CELL).contains(rule.name);
+
+    const expectedJson = JSON.parse(rule.actions.connectors[0].document);
+    cy.request({
+      method: 'GET',
+      url: `${Cypress.env('ELASTICSEARCH_URL')}/${rule.actions.connectors[0].index}/_search`,
+      headers: { 'kbn-xsrf': 'cypress-creds' },
+    }).then((response) => {
```

Comment:
There could be a gap before indexed documents become searchable. If the request has failed or returned no results, we should retry it without retrying the whole test.

It would generally be better to verify logic like that (that a document has been indexed) with integration tests. e2e tests are more suitable for verifying user interactions, e.g., that rule actions and connectors are editable, and after editing, their state persisted. What do you think about splitting this test into an e2e + integration to cover the rule actions completely?

Reply:
In this PR I've implemented what I understood by @banderror requirements. We can continue expanding the coverage. With this one, we are verifying a complete flow for rule actions.

Reply:
> There could be a gap before indexed documents become searchable. If the request has failed or returned no results, we should retry it without retrying the whole test.

Seems like @MadameSheema addressed this concern:

```ts
    /* Once the action is triggered we wait for the new document to be indexed */
    waitForNewDocumentToBeIndexed(index, initialNumberofDocuments);

    /* We assert that the new indexed document is the one set on the index action */
    cy.request({
      method: 'GET',
      url: `${Cypress.env('ELASTICSEARCH_URL')}/${index}/_search`,
      headers: { 'kbn-xsrf': 'cypress-creds' },
    }).then((response) => {
      expect(response.body.hits.hits[1]._source).to.deep.equal(expectedJson);
    });
```

Reply:
> In this PR I've implemented what I understood by @banderror requirements. We can continue expanding the coverage. With this one, we are verifying a complete flow for rule actions.

👍 I'd say let's keep this e2e test, I think it's useful. As for expanding the coverage at the integration level, I will create a technical ticket for that. We could test a lot more than just indexing a simple document via the index connector.